harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -738,6 +777,33 @@ def build_validation_data_loader(self) -> DataLoader:
         """
         pass
 
+    def create_lr_scheduler(
+        self, optimizer: torch.optim.Optimizer  # type: ignore
+    ) -> Optional[LRScheduler]:
+        """
+        Create a learning rate scheduler for the trial given an instance of the
+        optimizer.
+
+        Arguments:
+            optimizer (torch.optim.Optimizer): instance of the optimizer to be
+                used for training
+
+        Returns:
+            :py:class:`det.pytorch.LRScheduler`:
+                Wrapper around a :obj:`torch.optim.lr_scheduler._LRScheduler`.
+        """
+        pass
+
+    def build_callbacks(self) -> Dict[str, _callback.PyTorchCallback]:
```

Comment:
See `_save()` implementation -- we need a way to link the restored callback state to the correct callback, if there are multiple. If we used a list, we would have to depend on the order of the callback list which would get messy.

Reply:
If we keep it as a list why would would it get messy, since the return of `build_callbacks` is immutable across pause and restart? I just find the return type `Dict[str, _callback.PyTorchCallback]` a bit strange.